### PR TITLE
User can log in to the application

### PR DESCRIPTION
**Pull request by NiciP**
- Happy path created 
- Sad path created for email / password

### CODE REVIEW

File: features/step_definitions/basic_steps.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+  When ("I visit the site") do
+    visit root_path
+  end
+  
+  Then ("I should see {string}") do |content|
+    expect(page).to have_content content
+  end
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: You should get into the habit of placing your assertions in a separate file

File: features/step_definitions/basic_steps.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+   fill_in field, with: input
+  end
+  
+  When("I click on {string}") do |string|
+    click_on string
+  end
+  
+  Given("following users exist") do |table|
+    table.hashes.each do |user|
+        User.create!(user)
```
Comment (tochman) [inline features/step_definitions/basic_steps.rb]: I would like you to look into using factories rather than creating "real" instances

**Review by amandagibson — APPROVED** [review]

**Review by yanqiuzhang — APPROVED** [review]

File: features/step_definitions/basic_steps.rb (hunk 1 above)
Comment (NiciP) [reply features/step_definitions/basic_steps.rb]: Okay, I will create assertions.rb file and move those lines of code accordingly.

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (NiciP) [reply features/step_definitions/basic_steps.rb]: We will work on this tomorrow morning asap, thanks

Comment (NiciP) [reply features/step_definitions/basic_steps.rb]: Implemented factory "Test" for user

**Review by vick3d — APPROVED** [review]

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (faraznaeem) [reply features/step_definitions/basic_steps.rb]: You need to implement the Factory here in the step as well

File: db/schema.rb [hunk 3]
```diff
@@ -65,6 +65,13 @@
     t.index ["receiver_id", "receiver_type"], name: "index_mailboxer_receipts_on_receiver_id_and_receiver_type"
   end
 
+  create_table "tests", force: :cascade do |t|
+    t.string "email"
+    t.string "password"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+  end
+
```
Comment (faraznaeem) [inline db/schema.rb]: Why is this created?

File: db/migrate/20190213083129_create_tests.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class CreateTests < ActiveRecord::Migration[5.1]
+  def change
+    create_table :tests do |t|
+      t.string :email
+      t.string :password
+
+      t.timestamps
+    end
+  end
```
Comment (faraznaeem) [inline db/migrate/20190213083129_create_tests.rb]: Why did you create a migration for this feature?

File: features/assertions.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+Then ("I should see {string}") do |content|
+    expect(page).to have_content content
+  end
+
```
Comment (faraznaeem) [inline features/assertions.rb]: this file should be named assertions_steps.rb

File: features/step_definitions/basic_steps.rb (hunk 2 above)
Comment (NiciP) [reply features/step_definitions/basic_steps.rb]: Ok done

File: features/assertions.rb (hunk 5 above)
Comment (NiciP) [reply features/assertions.rb]: This is corrected

File: db/schema.rb (hunk 3 above)
Comment (NiciP) [reply db/schema.rb]: Incorrectly created a second database, dropped it now and migrated db again

File: db/migrate/20190213083129_create_tests.rb (hunk 4 above)
Comment (NiciP) [reply db/migrate/20190213083129_create_tests.rb]: I think this is related the incorrect creation of second db as above

**Review by faraznaeem — APPROVED** [review]

**Pull request merged**